qubes-rpc/tar2qfile.c

```diff
@@ -762,7 +764,9 @@ ustar_rd (int fd, struct file_header * untrusted_hdr, char *buf, struct stat * s
 			dirs_headers_sent[n_dirs-1] = malloc(sizeof (char) * (strlen(path)+1));
 			if (dirs_headers_sent[n_dirs-1] == NULL)
 				return MEMORY_ALLOC_FAILED;
-			strncpy(dirs_headers_sent[n_dirs-1], path, strlen(path)+1);
+			
+			memcpy(dirs_headers_sent[n_dirs-1], path, strlen(path)+1);
+			dirs_headers_sent[n_dirs+strlen(path)] = '\0';
```

Comment:
First, if you copy `strlen(path)+1` bytes, then it already include final `\0`. Second `dirs_headers_sent[n_dirs+strlen(path)]` is wrong - `dirs_headers_sent` is an array of strings, not just a string, so there should be two indexes. But because of the first point, this line is unnecessary.